samcli/lib/providers/sam_base_provider.py

```diff
@@ -61,14 +68,28 @@ def _extract_lambda_function_code(resource_properties: Dict, code_property_key:
         str
             Representing the local code path
         """
-
         codeuri = resource_properties.get(code_property_key, SamBaseProvider.DEFAULT_CODEURI)
 
         if isinstance(codeuri, dict):
             return SamBaseProvider.DEFAULT_CODEURI
 
         return cast(str, codeuri)
 
+    @staticmethod
+    def _is_code_uri_s3(codeuri: Optional[Any]) -> bool:
+        return (isinstance(codeuri, dict) and ("S3Bucket" in codeuri or "Bucket" in codeuri)) or (
+            isinstance(codeuri, str) and codeuri.startswith("s3://")
+        )
+
+    @staticmethod
+    def _warn_code_extraction(resource_type: str, resource_name: str, code_property: str) -> None:
+        LOG.warning(
+            "%s '%s' has specified S3 location for %s which is unsupported.",
```

Comment:
This warning feels a bit ambiguous here. Shouldn't we specify the scope of this "unsupported"?